python-sdk/nuimages/nuimages.py

```diff
@@ -701,52 +794,58 @@ def render_image(self,
         im = Image.open(im_path)
 
         # Initialize drawing.
-        font = ImageFont.load_default()
+        font = get_font(fonts_valid=['FreeSerif.ttf', 'FreeSans.ttf'], font_size=font_size)
         draw = ImageDraw.Draw(im, 'RGBA')
 
-        if with_annotations:
-            # Load stuff / background regions.
-            surface_anns = [o for o in self.surface_ann if o['sample_data_token'] == sd_token_camera]
-            if surface_tokens is not None:
-                surface_anns = [o for o in surface_anns if o['token'] in surface_tokens]
-
-            # Draw stuff / background regions.
-            for ann in surface_anns:
-                # Get color and mask.
-                category_token = ann['category_token']
-                category_name = self.get('category', category_token)['name']
-                color = self.color_map[category_name]
-                if ann['mask'] is None:
-                    continue
-                mask = mask_decode(ann['mask'])
-
-                # Draw mask. The label is obvious from the color.
-                draw.bitmap((0, 0), Image.fromarray(mask * 128), fill=tuple(color + (128,)))
-
-            # Load object instances.
-            object_anns = [o for o in self.object_ann if o['sample_data_token'] == sd_token_camera]
-            if object_tokens is not None:
-                object_anns = [o for o in object_anns if o['token'] in object_tokens]
-
-            # Draw object instances.
-            for ann in object_anns:
-                # Get color, box, mask and name.
-                category_token = ann['category_token']
-                category_name = self.get('category', category_token)['name']
-                color = self.color_map[category_name]
-                bbox = ann['bbox']
-                attr_tokens = ann['attribute_tokens']
-                attributes = [self.get('attribute', at) for at in attr_tokens]
-                name = annotation_name(attributes, category_name, with_attributes=with_attributes)
-                if ann['mask'] is None:
-                    continue
-                mask = mask_decode(ann['mask'])
-
-                # Draw mask, rectangle and text.
-                draw.bitmap((0, 0), Image.fromarray(mask * 128), fill=tuple(color + (128,)))
-                draw.rectangle(bbox, outline=color, width=box_line_width)
-                if with_category:
-                    draw.text((bbox[0], bbox[1]), name, font=font)
+        with_annotations_options = ['all', 'surfaces', 'objects', 'none']
+        assert with_annotations in with_annotations_options, \
+            'Error: {} is not a valid option for with_annotations. ' \
```

Comment:
maybe raise NotImplementedError is better

Reply:
no i think this is fine, this is already an assertion error

Reply:
But that sounds like it is not implemented yet. It is simply a wrong input.
@whyekit-aptiv Also `with_annotations` sounds like it's a bool

Reply:
Changed from `with_annotations` to `annotations_type`